play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/implicits/RichDateInputSupport.scala

```diff
@@ -65,35 +145,72 @@ trait RichDateInputSupport {
     private[views] def withId(field: Field): DateInput =
       withStringProperty(field.name, dateInput.id, dateInput)((dateInput, id) => dateInput.copy(id = id))
 
-    private[views] def withInputItems(field: Field): DateInput = {
+    private[views] def deprecatedWithInputItems(field: Field): DateInput = {
+      val dateInputItems: Seq[InputItem] =
+        if (dateInput.items.size == 3) dateInput.items else Seq.fill(3)(InputItem.defaultObject)
+
+      val items = Seq(
+        deprecatedInputItem(field, dateInputItems(0), "day", className = "govuk-input--width-2"),
+        deprecatedInputItem(field, dateInputItems(1), "month", className = "govuk-input--width-2"),
+        deprecatedInputItem(field, dateInputItems(2), "year", className = "govuk-input--width-4")
+      )
+
+      dateInput.copy(items = items)
+    }
+
+    private def deprecatedInputItem(field: Field, inputItem: InputItem, key: String, className: String): InputItem = {
       def errorClass(itemField: Field) =
         if (field.errors.nonEmpty || itemField.errors.nonEmpty) "govuk-input--error" else ""
 
-      def inputItem(inputItem: InputItem, key: String, className: String): InputItem = {
-        val defaultInputItem = InputItem.defaultObject
-        val classes          = if (inputItem.classes == defaultInputItem.classes) className else inputItem.classes
+      val defaultInputItem = InputItem.defaultObject
+      val classes          = if (inputItem.classes == defaultInputItem.classes) className else inputItem.classes
 
-        inputItem.copy(
-          id = if (inputItem.id == defaultInputItem.id) s"${field.name}.$key" else inputItem.id,
-          name = if (inputItem.name == defaultInputItem.name) s"${field.name}.$key" else inputItem.name,
-          value = if (inputItem.value == defaultInputItem.value) field(key).value else inputItem.value,
-          label =
-            if (inputItem.label == defaultInputItem.label) Some(messages(s"date.input.$key")) else inputItem.label,
-          classes = s"$classes ${errorClass(field(key))}".trim
-        )
-      }
+      inputItem.copy(
+        id = if (inputItem.id == defaultInputItem.id) s"${field.name}.$key" else inputItem.id,
+        name = if (inputItem.name == defaultInputItem.name) s"${field.name}.$key" else inputItem.name,
+        value = if (inputItem.value == defaultInputItem.value) field(key).value else inputItem.value,
+        label = if (inputItem.label == defaultInputItem.label) Some(messages(s"date.input.$key")) else inputItem.label,
+        classes = s"$classes ${errorClass(field(key))}".trim
+      )
+    }
 
-      val dateInputItems = if (dateInput.items.size == 3) dateInput.items else Seq.fill(3)(InputItem.defaultObject)
+    private def defaultInputItem(field: Field, key: String, className: String): InputItem = {
+      def errorClass(itemField: Field) =
+        if (field.errors.nonEmpty || itemField.errors.nonEmpty) "govuk-input--error" else ""
```

Comment:
I think we've got this same bit in two places now, it could be hoisted up a level as a private def

Reply:
@oscarduignan On this one, we have similar but not identical methods for `deprecatedInputItem` and `defaultInputItem`. In my head I split them out because otherwise we have to munge in some unnecessary stuff - but happy to talk this one through?

Reply:
that makes sense! save future accidents